Subcaption package example

keaganbester opened the pull request:
The samplefigure python script has been modified to generate a few images that are then pulled into the report template.

If the python script's increase in work is considered redundant, I can modify the .tex file to replicate the original sample figure several times instead. 

The content of the figures themselves are not spectacular, but the noise factor was the simplest way to iterate the current set of images. If a more useful set of images is preferred and the extra python work time is allowed,  I am sure I can find something in the dynamics and control notebooks that I can add instead.

keaganbester committed:
Merge pull request #5 from ChemEngUP/master

Merged with master

keaganbester committed:
Merge pull request #6 from ChemEngUP/master

Merging to base

keaganbester committed:
An example using the subcaption package

An example of creating a figure with multiple subfigures and a subcaption for each subfigure

alchemyst commented on report_template.tex:
```diff
@@ -134,6 +134,43 @@ \section{Results and Discussion}
+\begin{figure}
+	\begin{subfigure}[b]{.33\textwidth}
+		\centering\includegraphics[width=\textwidth]{modelfigure_0.pdf}
+		\caption{noise factor = 1.0}\label{fig:1a}
```
Better to capitalise this properly

keaganbester replied:
Corrected, that was sloppy on my part

alchemyst commented on report_template.tex:
```diff
@@ -134,6 +134,43 @@ \section{Results and Discussion}
+		\caption{noise factor = 1.0}\label{fig:1a}
+	\end{subfigure}\hfill
+	\begin{subfigure}[b]{.33\textwidth}
+		\centering\includegraphics[width=\textwidth]{modelfigure_1.pdf}
```
Probably better practice to encode the value of the noise factor in the filename somehow

keaganbester replied:
Encoded using the value of the noise factor instead of the enumerated index. I've replaced the decimal point with a "p" to ensure there are no filename extension issues. Originally I converted the base 10 decimal to floating-point hexadecimal but at 32 bits long the filename was a little ridiculous.

alchemyst replied:
Agreed about the 32 bits!

alchemyst commented on samplefigure.py:
```diff
@@ -23,3 +23,31 @@
 plt.savefig('samplefigure.pdf')
+
+#Lets investigate the effect of noise on our Model "A" and it's derivative
+for index, value  in enumerate([1.0, 10, 100]):
```
Good idea to generate all these files seperately, but we really should be using a function to do the decorations and plotting rather than copy/pasting all these blocks. I want to model best practice.

keaganbester replied:
I struggled to figure out how far to go when creating these functions, I've made 2 functions and while they aren't universal in their application, they do make the for-loop only 4 lines long

alchemyst replied:
I'm OK with it as it stands right now. We can add a refactor to the issue list.

alchemyst reviewed (changes requested):
I'd prefer that we model best practice with our examples

keaganbester committed:
Subcaption example, revised

Changes to the samplefigure.py to attempt best practices through the use of functions to generate a graph and to style the graph

alchemyst commented:
@keaganbester you can keep the pull request open if you plan to commit changes that address my comments. Closing indicates that you're not going to pursue the changes.

keaganbester commented:
@alchemyst sorry I wasn't entirely sure how to edit my existing pull request to address the changes, I've since created a new pull request but I realise that this isn't the correct practice.

alchemyst commented:
The workflow would simply be to make the changes and push. The new commits will automatically be recognised and added to the pull request. I'd rather you reopen this request and close the other one so that we can retain continuity.

alchemyst commented on report_template.tex:
```diff
@@ -134,6 +134,40 @@ \section{Results and Discussion}
+		\centering\includegraphics[width=\textwidth]{differentialfigure_100.pdf}
+		\caption{Noise factor = 100}\label{fig:1f}
+	\end{subfigure}
+	\caption[Short caption which will be in the table of figures]{Example of a figure that contains multiple subfigures.  For more options on using subfigures and more complicated subcaptions, see the package documentation at \url{https://gitlab.com/axelsommerfeldt/caption} To update this figure, run samplefigure.py and recompile the document.}
```
Doesn't this clash with the other caption?

alchemyst replied:
Specifically the short caption in square brackets.

keaganbester replied:
I kept it as a placeholder for template purposes, but I can make it unique without a problem.

alchemyst commented on samplefigure.py:
```diff
@@ -18,8 +77,23 @@
+diffB = numpy.diff(modelB)
+dx = x[1]-x[0]
+
+for index, value in enumerate([1.0, 10, 100]):
```
We don't use index in this loop do we?

keaganbester replied:
Not any more no, it can be simplified down to
`for value in [1, 10, 100]:`
